scio-extra/src/main/scala/com/spotify/scio/extra/Sparkey.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.extra
+
+import java.io._
+import java.net.URI
+import java.nio.ByteBuffer
+import java.nio.file.{Paths, Files => JFiles}
+
+import com.google.common.io.Files
+import com.spotify.scio.ScioContext
+import com.spotify.scio.values.{SCollection, SideInput}
+import com.spotify.sparkey.{CompressionType, SparkeyReader, SparkeyWriter, Sparkey => JSparkey}
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions
+import org.apache.beam.sdk.options.PipelineOptionsFactory
+import org.apache.beam.sdk.transforms.{DoFn, PTransform, View}
+import org.apache.beam.sdk.util.GcsUtil.GcsUtilFactory
+import org.apache.beam.sdk.util.gcsfs.GcsPath
+import org.apache.beam.sdk.values.{PCollection, PCollectionView}
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object Sparkey {
+
+  implicit class SparkeyScioContext(val self: ScioContext) {
+    def sparkeySideInput(url: SparkeyUri): SideInput[SparkeyReader] = {
+      val view = self.parallelize(Seq(url)).applyInternal(sparkeyTransform)
+      new SparkeySideInput(view)
```

Comment:
Do you really need `sparkeyTransform`? `self.internal.apply(View.asSingleton)` should just work?